Gemfile.lock

```diff
@@ -74,7 +75,9 @@ GEM
       minitest (>= 5.1)
       tzinfo (~> 2.0)
     ast (2.4.2)
+    brotli (0.4.0)
```

Comment:
I don't feel great about adding this as a dependency...

1. There are [known segfaults in the issue tracker that have been left unaddressed for a year+](https://github.com/miyucy/brotli/issues/42)
2. There does not seem to be any testing with `GC.stress = true` to ensure GC safety

Essentially, it seems like we'd be inheriting a bunch of buggy, questionably maintained C code...

Reply:
any alternatives you'd recommend?

Reply:
Correct me if I'm wrong, but those segfaults are just type checking on the parameters. Since we control this side, is there a risk? Or is your concern more broad stability?

Reply:
At risk of sounding Rust-pilled, it's more of a general concern about C code that we consider safe to adopt widely in production. At a bare minimum, I think we should set some basic rules:

1. Should be tested against latest stable Ruby ([this gem only tests up to 3.0](https://github.com/miyucy/brotli/blob/66cfbde645ee842f92c8b3e5abc7b3feebeda527/.github/workflows/main.yml#L10))
2. Leverages [`ruby_memcheck`](https://github.com/Shopify/ruby_memcheck) to test for memory issues
3. Runs test suite with `GC.stress = true` to smoke out GC related issues before they make it into production

FWIW, all of the above applies to _all_ native extensions, both C and Rust.

Reply:
Well, the fix was merged in less than 20 minutes, so there's that.